test/test_distributions.py

```diff
@@ -245,6 +245,19 @@ def _check_sampler_sampler(self, torch_dist, ref_dist, message, multivariate=Fal
             self.assertLess(-threshold, bias, message)
             self.assertLess(bias, threshold, message)
 
+    def _check_sampler_discrete(self, torch_dist, ref_dist, message,
+                                num_samples=10000, failure_rate=1e-3):
+        """Runs a Chi2-test for the support, but ignores tail instead of combining"""
+        samples = torch_dist.sample_n(num_samples).squeeze()
+        if isinstance(torch_samples, Variable):
+            torch_samples = torch_samples.data
+        torch_samples = torch_samples.cpu().numpy()
+        unique, counts = np.unique(torch_samples, return_counts=True)
+        pmf = ref_dist.pmf(unique)
+        msk = (counts > 5) & ((pmf * num_samples) > 5)
+        chisq, p = chisquare(counts[msk], pdf[msk] * NUM_SAMPLES)
```

Comment:
Nice. Where can I read about how to use the `chisquare` function?

I've sometimes found is necessary to bin items into the top 10 categories plus one category for everything else. But maybe this `chisquare` function is less susceptible to low-probability classes than the simple Pearson's chi^2 test I was using.

Reply:
It's actually just the Pearson's chi^2 test as well, but NIST [says](http://www.itl.nist.gov/div898/handbook/eda/section3/eda35f.htm) that as long as the expected frequency and actual counted frequency are over 5, it should be fine. I've plotted the p-values and they seem normally distributed both when using the Scipy samplers and our samplers.

Reply:
Ok, just beware that there will likely be bins with fewer than 5 points; if this is a problem, we can simply merge small bins.

Reply:
Right, I'm currently masking those bins for now, using `msk` above. We can merge bins too, but this was simpler to do at a first pass.